releases/release-5.0.4.md

```diff
@@ -0,0 +1,165 @@
+---
+title: TiDB 5.0.4 Release Notes
+---
+
+# TiDB 5.0.4 Release Notes
+
+Release Date: September 27, 2021
+
+TiDB version: 5.0.4
+
+## Compatibility Changes
+
++ TiDB
+
+    - Revert #19341 to avoid #24326 [#26258](https://github.com/pingcap/tidb/pull/26258)
```

Comment:
- TiDB
  - Revert #19341 to avoid #24326 [#26258](https://github.com/pingcap/tidb/pull/26258)
  - Fix the case that TiDB would return the wrong result when the children of the UNION contain pure NULL values [#26571](https://github.com/pingcap/tidb/pull/26571)
  - Fix the issue that greatest(datetime) union null returns empty string [#26565](https://github.com/pingcap/tidb/pull/26565)
  - fix incompatible last_day func behavior in sql mode [#26000](https://github.com/pingcap/tidb/pull/26000)
  - fix wrong selection push down when having above agg [#27742](https://github.com/pingcap/tidb/pull/27742)
  - fix expression rewrite makes between expr infers wrong collation. [#27548](https://github.com/pingcap/tidb/pull/27548)
  - make `group_concat` function consider the collation [#27528](https://github.com/pingcap/tidb/pull/27528)
  - Fix bug that count disctinct on multi-columns return wrong result when new collation is on. [#27507](https://github.com/pingcap/tidb/pull/27507)
  - fix extract bug when argument is a negative duration [#27368](https://github.com/pingcap/tidb/pull/27368)
  - fix unexpected behavior when casting invalid string to date [#27110](https://github.com/pingcap/tidb/pull/27110)
  - Fix an issue that NO_ZERO_IN_DATE does not work on the default values. [#26903](https://github.com/pingcap/tidb/pull/26903)
  - fix a bug on the query range of prefix index [#26261](https://github.com/pingcap/tidb/pull/26261)
  - fix load data with non-utf8 can succeed [#26143](https://github.com/pingcap/tidb/pull/26143)
  - Fix the wrong result after "insert ignore on duplicate update" in the secondary index has the same column in primary key but has prefix length [#25905](https://github.com/pingcap/tidb/pull/25905)
  - Fix the wrong result for "insert ignore duplicate up" on partition table when handle changed and duplicated on secondary index [#25859](https://github.com/pingcap/tidb/pull/25859)
  - fix wrong enum key in point get [#24772](https://github.com/pingcap/tidb/pull/24772)
  - fix wrong flen infer for bit constant [#24267](https://github.com/pingcap/tidb/pull/24267)
  - fix get var expr when session var is hex literal [#23373](https://github.com/pingcap/tidb/pull/23373)
  - fix unexpected constant fold when year compare string. [#23336](https://github.com/pingcap/tidb/pull/23336)
  - Enlarge the variable tidb_stmt_summary_max_stmt_count default value from 200 to 3000 [#25873](https://github.com/pingcap/tidb/pull/25873)

Reply:
Note: overwrite the original release notes in this PR.